quint-assign1/quint-assign1.py

```diff
@@ -0,0 +1,218 @@
+"""
+Assignment 1: Cryptography
+Course: CS 92SI
+Name: Quint Underwood
+Date: 9 October 2015
+
+Replace this with a description of the program.
+"""
+
+def encrypt_caesar(plaintext):
+    """
+    Encrypts plaintext using a Caesar cipher.
+    Add more implementation details here.
+    """
+    shift = int(input("By how many letters do you want to shift? "))
+    encoded_phrase = ""
+    for letter in plaintext:
+        letter = chr(ord(letter) + shift)
+        if letter > 'z':
+            letter = chr(ord(letter) - 26)
+        encoded_phrase += letter
+    #print(encoded_phrase.upper())
+    return(encoded_phrase.upper())
+
+
+def decrypt_caesar(ciphertext):
+    """
+    Decrypts a ciphertext using a Caesar cipher.
+    Add more implementation details here.
+    """
+    shift = int(input("How many letters was this phrase shifted when it was encoded? "))
+    decoded_phrase = ""
+    for letter in ciphertext:
+        letter = chr(ord(letter) - shift)
+        if letter < 'a':
+            letter = chr(ord(letter) + 26)
+        decoded_phrase += letter
+    return(decoded_phrase.upper())
+
+def encrypt_vigenere(plaintext, keyword):
+    """
+    Encrypts plaintext using a Vigenere cipher with a keyword.
+    Add more implementation details here.
+    """
+    encoded_phrase = ""
+    length = len(keyword)
+    i=0
+    for letter in plaintext:
+        letter = chr(ord(letter) + ord(keyword[i%length]) - ord('a'))
+        if letter > 'z':
+            letter = chr(ord(letter) - 26)
+        encoded_phrase += letter
+        i += 1
+    return (encoded_phrase)
+
+
+
+def decrypt_vigenere(ciphertext, keyword):
+    """
+    Decrypts ciphertext using a Vigenere cipher with a keyword.
+    Add more implementation details here.
+    """
+    decoded_phrase = ""
+    length = len(keyword)
+    i=0
+    for letter in ciphertext:
+        letter = chr(ord(letter) - ord(keyword[i%length]) + ord('a'))
+        if letter < 'a':
+            letter = chr(ord(letter) + 26)
+        decoded_phrase += letter
+        i += 1
+    return (decoded_phrase)
+
+
+def encrypt_railfence(plaintext, num_rails):
+    """
+    Encrypts plaintext using a railfence cipher.
+    Add more implementation details here.
+    """
+    divisor = 2 * num_rails - 2
+    encoded_phrase = ""
+    rails = [[letter for n, letter in enumerate(plaintext) if ((n+i) % divisor == 0 or (n-i) % divisor == 0)] for i in range(num_rails)]
+    for rail in rails:
+        for element in rail:
+            encoded_phrase += element
```

Comment:
You can actually simplify this with a nested list comprehension:
`encoded_phrase = [elem for elem in rail for rail in rails]